### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [3]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper functions

### Description

Added documentation to following files:
- [x] exam/ExamFactory.java
- [x] exam/ExamUtilService.java

### Steps for Testing
_JavaDoc changes only_

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

**Review by laadvo — CHANGES_REQUESTED** [review]
There are still some helper functions in `ExamUtilService` without JavaDoc, for example `addExercisesWithParticipationsAndSubmissionsToStudentExam`. Please add JavaDoc to where it is missing

**Review by b-fein — APPROVED** [review]

File: src/test/java/de/tum/in/www1/artemis/exam/ExamUtilService.java [hunk 1]
```diff
@@ -470,24 +635,52 @@ public ExamSession addExamSessionToStudentExam(StudentExam studentExam, String s
         return examSession;
     }
 
-    public StudentExam addStudentExamForActiveExamWithUser(String user) {
+    /**
+     * Creates and saves a StudentExam with a registered User. Also creates and saves an empty course and an Exam that started 1 hour ago and ends in 1 hour.
```
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/exam/ExamUtilService.java:639]: I think you shoud reformulate this to say that first a course and an exam (using the course and student users) are created and this exam is then used to create a student exam.

**Review by laadvo — APPROVED** [review]
LGTM, just a small suggestion

File: src/test/java/de/tum/in/www1/artemis/exam/ExamUtilService.java (hunk 1 above)
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/exam/ExamUtilService.java]: reformulated 👍

**Pull request merged**